File: backend/apps/backend/src/user/api-keys/api-keys.controller.ts
import { ApiKeyDto, UpdateApiKeyDto } from '@app/dto';
import { Body, Controller, Get, Post, Req, UseGuards, UsePipes, ValidationPipe } from '@nestjs/common';
import { JwtAuthGuard } from '../../auth/token/access-token.guard';
import { ApiKeysService } from './api-keys.service';

@UsePipes(new ValidationPipe())
@UseGuards(JwtAuthGuard)
@Controller('user/api-keys')
export class ApiKeysController {
  constructor(private readonly apiKeysService: ApiKeysService) {}

  @Get()
  async getAll(@Req() request: Express.Request) {
    return await this.apiKeysService.getAll(request.user['_id'])
  }

  @Post('create')
  async create(@Body() dto: ApiKeyDto, @Req() request: Express.Request) {
    return await this.apiKeysService.create(request.user['_id'], dto)
  }

  @Post('update')
  async update(@Body() dto: UpdateApiKeyDto, @Req() request: Express.Request) {
    return await this.apiKeysService.update(request.user['_id'], dto)
  }

  @Post('refresh')
  async refresh(@Body('apiKey') apiKey: string, @Req() request: Express.Request) {
    return await this.apiKeysService.refresh(request.user['_id'], apiKey)
  }

  @Post('delete')
  async delete(@Body('apiKey') apiKey: string, @Req() request: Express.Request) {
    return await this.apiKeysService.delete(request.user['_id'], apiKey)
  }
}

File: backend/Backend.Api/Controllers/User/Dropshipper/ApiKeysController.cs
using Backend.Core.DTOs.User.Dropshipper.AllowHost;
using Backend.Core.DTOs.User.Dropshipper.ApiKey;
using Backend.Core.Entities;
using Backend.Core.Services.User.Dropshipper;
using backend.Extensions;
using Microsoft.AspNetCore.Mvc;

namespace backend.Controllers.User.Dropshipper;

[Route("api/user/api-keys")]
public class ApiKeysController : Controller
{
    private readonly ApiKeysService _apiKeysService;

    public ApiKeysController(ApiKeysService apiKeysService)
    {
        _apiKeysService = apiKeysService;
    }

    [HttpGet]
    public async Task<IEnumerable<ApiKeyDto>> GetAll()
    {
        return await _apiKeysService.GetAll(User.Identity!.GetId());
    }
    
    [HttpPost("create")]
    public async Task Create([FromBody] CreateApiKeyDto dto)
    {
        await _apiKeysService.Create(User.Identity!.GetId(), dto);
    }
    
    [HttpPost("update")]
    public async Task Update([FromBody] UpdateApiKeyDto dto)
    {
        await _apiKeysService.Update(User.Identity!.GetId(), dto);
    }
    
    [HttpPost("delete")]
    public async Task Delete([FromBody] DeleteApiKeyDto dto)
    {
        await _apiKeysService.Delete(User.Identity!.GetId(), dto);
    }
    
    [HttpPost("refresh")]
    public async Task<ApiKeyDto> Refresh([FromBody] RefreshApiKeyDto dto)
    {
        return await _apiKeysService.Refresh(User.Identity!.GetId(), dto);
    }
}